packages/valory/skills/abstract_round_abci/test_tools/integration.py

```diff
@@ -169,10 +169,7 @@ def process_message_cycle(
         :param mining_interval_secs: the mining interval used in the tests
         :return: the incoming message
         """
-        if (
-            expected_types is not None
-            and tuple(expected_types.keys())[0] == "transaction_receipt"
-        ):
+        if expected_types and tuple(expected_types)[0] == "transaction_receipt":
```

Comment:
this is a fix in the test_tools/integration for which I did not waste time writing a test first. It should be evident that an empty dict would result in an `IndexError`, and now no longer will.